action.yml

```diff
@@ -0,0 +1,60 @@
+name: 'NPM Version Incrementer'
+description: 'Increments the version of an NPM package based on the branch.'
+outputs:
+  branch-name:
+    description: 'The current branch name.'
+    value: ${{ steps.get-branch-name.outputs.branch-name }}
+  current-version:
+    description: 'The current version of the package.'
+    value: ${{ steps.get-current-version.outputs.current-version }}
+  new-version:
+    description: 'The new version of the package.'
+    value: ${{ steps.get-new-version.outputs.new-version }}
+runs:
+  using: "composite"
+  steps:
+    - id: get-branch-name
+      shell: bash
+      run: |
+          # Get the current branch name
+          BRANCH=$(git branch --show-current)
+          echo "Current branch is - $BRANCH"
+          echo "branch-name=$BRANCH" >> $GITHUB_OUTPUT
+
+    - id: get-current-version
+      shell: bash
+      run: |
+          # Get the current version number
+          CURRENT_VERSION=$(jq -r '.version' package.json)
+          echo "Current version is - $CURRENT_VERSION"
+          echo "current-version=$CURRENT_VERSION" >> $GITHUB_OUTPUT
+
+    - id: npm-version-increment
+      shell: bash
+      run: |
+          BRANCH=${{ steps.get-branch-name.outputs.branch-name }}
```

Comment:
Maybe also add another output, like a boolean one if this increment was a patch. So the consumer of this actions doesn't have to calculate it again.

Reply:
So it would be true if the version change is a patch, and false if it's a major/minor version?

Reply:
yes

Reply:
or you can have a string variable to say the version change: 'major', 'minor' or 'patch'.